detox/test/e2e/05.waitfor.test.js

```diff
@@ -38,6 +37,11 @@ describe('WaitFor', () => {
     await expect(element(by.text('Text5'))).toBeVisible();
   });
 
+  it('should fail test after waiting for element to exist but it doesn\'t at the end', async () => {
+    await expect(element(by.id('neverAppearingText'))).toNotExist();
+    await expectToThrow(() => waitFor(element(by.id('neverAppearingText'))).toExist().withTimeout(1000));
```

Comment:
I think that equivalently, there should be an `expectToThrow()` over the `should wait until an element is removed` test as well